e2e/plugin-e2e/various-suite/mysql.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { selectors } from '@grafana/e2e-selectors';
+import { expect, test } from '@grafana/plugin-e2e';
+
+import {
+  tablesResponse,
+  fieldsResponse,
+  datasetResponse,
+  normalTableName,
+  tableNameWithSpecialCharacter,
+} from './mocks/mysql.mocks';
+
+test.beforeEach(async ({ context, selectors, explorePage }) => {
+  await explorePage.datasource.set('gdev-mysql');
+  await context.route(selectors.apis.DataSource.queryPattern, async (route, request) => {
+    switch (request.postDataJSON().queries[0].refId) {
+      case 'tables':
+        return route.fulfill({ json: tablesResponse, status: 200 });
+      case 'fields':
+        return route.fulfill({ json: fieldsResponse, status: 200 });
+      case 'datasets':
+        return route.fulfill({ json: datasetResponse, status: 200 });
+      default:
+        return route.continue();
+    }
+  });
+});
+
+test('code editor autocomplete should handle table name escaping/quoting', async ({ explorePage, selectors, page }) => {
+  await page.getByLabel('Code').check();
+  const editor = explorePage.getByTestIdOrAriaLabel(selectors.components.CodeEditor.container).getByRole('textbox');
+  await editor.fill('S');
+  await editor.press('ArrowDown+Enter');
+  await expect(page.getByLabel(tableNameWithSpecialCharacter)).toBeVisible();
+  await editor.press('Enter');
+
+  await expect(editor).toHaveValue(`SELECT  FROM grafana.\`${tableNameWithSpecialCharacter}\``);
+
+  for (let i = 0; i < tableNameWithSpecialCharacter.length + 2; i++) {
+    await editor.press('Backspace');
+  }
+
+  await editor.press('Control+I');
+  await expect(page.getByLabel(tableNameWithSpecialCharacter)).toBeVisible();
+});
+
+test('visual query builder should handle time filter macro', async ({ explorePage, page }) => {
+  await explorePage.getByTestIdOrAriaLabel('Table selector').click();
+  await page.getByText(normalTableName, { exact: true }).click();
+
+  // Open column selector
+  await explorePage.getByTestIdOrAriaLabel(selectors.components.SQLQueryEditor.selectColumn).click();
+  await page.getByText('createdAt').click();
+
+  // Toggle where row
+  await page.getByLabel('Filter').click();
+
+  // Click add filter button
+  await page.getByRole('button', { name: 'Add filter' }).click();
+  await page.getByRole('button', { name: 'Add filter' }).click(); // For some reason we need to click twice
+
+  // Open field selector
+  await explorePage.getByTestIdOrAriaLabel('Field').click();
+  await page.getByLabel('Select options menu').getByText('createdAt').click();
```

Comment:
`Select options menu` should be an e2e-selector. I'll make a note to fix that...